registry/data-models/transformation/models.py

```diff
@@ -0,0 +1,84 @@
+from registry.data-models.common.models import *
+from typing import Optional
+
+
+class SlidingWindowAggregationType(Enum):
+    """
+    Represents supported types of aggregation.
+    """
+    SUM = "sum"
+    COUNT = "count"
+    MAX = "maximum"
+    MIN = "minium"
+    AVG = "average"
+
+
+class SlidingWindowEmbeddingAggregationType(Enum):
+    """
+    Represents supported types for embedding aggregation.
+    Pooling is a sample-based discretization process. The objective is to down-sample an input
+    representation and reduce its dimensionality.
+    """
+    MAX_POOLING = "max_pooling"  # Max pooling is done by applying a max filter to (usually) non-overlapping subregions of the initial representation
+    MIN_POOLING = "min_pooling"  # Min pooling is done by applying a min filter to (usually) non-overlapping subregions of the initial representation.
+    AVG_POOLING = "avg_pooling"  # Average pooling is done by applying a average filter to (usually) non-overlapping subregions of the initial representation
+
+
+class WindowTimeUnit(Enum):
+    """
+     Represents a unit of time.
+    """
+    DAY = "day"
+    HOUR = "hour"
+    MINUTE = "minute"
+    SECOND = "second"
+
+
+class Window(BaseModel):
+    """
+    Represents a time window used in sliding window algorithms.
+    """
+    size: int  # Represents the duration of the window.
+    unit: WindowTimeUnit
+
+
+class SlidingWindowAggregation(Function):
+    """
+    Sliding window aggregation produces feature data by aggregating a collection of data within a given time
+    interval into an aggregate value. It ensures point-in-time correctness, when joining with label data,
+    it looks back the configurable time window from each entry's timestamp and compute the aggregate value.
+    This class can be extended to support LateralView in aggregation.
```

Comment:
@xiaoyongzhu. Scala code supports lateral view, even though current Python API for Frame Feature Definition does not. 

Do you want to still skip the lateral view in the Feature Registry data model too and add it one once the functionality is exposed in the Feature Definition API?